### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by sfc-gh-astus**

### CODE REVIEW

File: .github/dependabot.yml
Comment (sfc-gh-pjob) [inline .github/dependabot.yml:2]: I wonder if we shouldn't also configure dependabot for github actions (https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot).

@sfc-gh-astus , @sfc-gh-turbaszek WDYT?

I'm not very familiar with github actions, I've been using GitLab for last few years, but I see that we have already a few different versions of the same action (checkout) used in the pipeline configurations:
* checkout v1 & v2 -> https://github.com/Snowflake-Labs/snowcli/blob/main/.github/workflows/build.yaml
* checkout v3 -> https://github.com/Snowflake-Labs/snowcli/blob/main/.github/workflows/publish.yaml

Comment (sfc-gh-astus) [reply .github/dependabot.yml:2]: I think it is a good idea. I will add it.

About different versions in actions I think we should update them. I will create new PR with it.

Comment (sfc-gh-pjob) [reply .github/dependabot.yml:2]: > About different versions in actions I think we should update them. I will create new PR with it.

Hmm, I thought dependabot will help us with it :)

**Review by sfc-gh-pjob — APPROVED** [review]

File: .github/dependabot.yml
Comment (sfc-gh-pjob) [reply .github/dependabot.yml:2]: After private discussion with Adam I understood that a separate PR for GH actions upgrade is a good idea to avoid flooding with PRs containing initial upgrades created by dependabot separately for each action. 

I've also dismissed my previous PR approval because in our opinion we have to think about a similar PR for python dependencies upgrade before merging the dependabot configuration.

**Review by sfc-gh-jsikorski — APPROVED** [review]

**Review by sfc-gh-pjob — APPROVED** [review]
@sfc-gh-astus I've seen your comment about 5 PRs limit by default, so I'm reapproving this PR.

**Pull request merged**
